modules/javafx.base/src/main/java/javafx/beans/value/ObservableValue.java

```diff
@@ -251,4 +252,63 @@ default ObservableValue<T> orElse(T constant) {
     default <U> ObservableValue<U> flatMap(Function<? super T, ? extends ObservableValue<? extends U>> mapper) {
         return new FlatMappedBinding<>(this, mapper);
     }
+
+    /**
+     * Returns an {@code ObservableValue} that holds this value whenever the given
+     * condition evaluates to {@code true}, otherwise holds the last value when
+     * {@code condition} became {@code false}. The value is updated whenever this
+     * {@code ObservableValue} changes, unless the condition currently evaluates
+     * to {@code false}.
+     * <p>
+     * The returned {@code ObservableValue} only observes this value when the given
+     * {@code condition} evaluates to {@code true}. This allows this {@code ObservableValue}
+     * and the conditional {@code ObservableValue} to be garbage collected if neither is
+     * otherwise strongly referenced when {@code condition} becomes {@code false}.
+     * <p>
+     * A currently observed binding will observe its source, which means it will not be eligible
+     * for garbage collection while source isn't. However, using {@code when} this {@code ObservableValue}
```

Comment:
"while source isn't" -> "while **its** source"

"However, using `{@code when}` this `{@code ObservableValue}`" -> "However, when using `{@code when}`, this `{@code ObservableValue}`"